File: cinebook-frontend/src/services/movieService.js


import axiosInstance from "./axiosInstance";



export const getAllMovies = (page=0,size=5) => {
    return axiosInstance.get(`/api/movies?page=${page}&size=${size}`);
};

export const getMovieById = (id) => {
    return axiosInstance.get(`/api/movies/${id}`);
};
export const getShowsByMovie = (movieId) => {
    return axiosInstance.get(`/api/show/movie/${movieId}`);
};

export const searchMovie=(title)=>{
    return axiosInstance.get(`/api/movies/search/${title}`);
}
export const getMoviesByGenre=(genre)=>{
    return axiosInstance.get(`/api/movies/genre/${genre}`);
}

export const getRunningMovies=()=>{
    return axiosInstance.get("/api/movies/running");
}
export const getUpcomingMovies=()=>{
    return axiosInstance.get("/api/movies/upcoming");
}
export const getReleaseMovies=()=>{
    return axiosInstance.get("/api/movies/released");
}

export const getMoviesByLanguage=(language)=>{
    return axiosInstance.get(`/api/movies/language/${language}`);
}

export const getAllMovie=()=>{
    return axiosInstance.get("/api/movies/all");
}



//for admin dashboard

export const addMovie=(movie)=>{
    return axiosInstance.post("/api/movies",movie);
}
export const updateMovies=(id,movie)=>{
    return axiosInstance.put(`/api/movies/${id}`,movie);
}

export const activateMovie=(id)=>{
    return axiosInstance.put(`/api/movies/actMovie/${id}`);
}
export const deleteMovie=(id)=>{
    return axiosInstance.put(`/api/movies/deactMovie/${id}`);

}

File: cinebook-backend/cinebook-backend/src/main/java/com/cinebook/cinebook/service/MovieService.java
package com.cinebook.cinebook.service;

import com.cinebook.cinebook.dto.request.MovieRequestDto;
import com.cinebook.cinebook.dto.response.MovieResponseDto;
import com.cinebook.cinebook.entity.Movie;
import org.springframework.data.domain.Page;

import java.time.LocalDate;
import java.util.List;
import java.util.Optional;

public interface MovieService {

    //CRUD
    MovieResponseDto addmovie(MovieRequestDto movieRequestDto);//add the movie
    Page<MovieResponseDto>getAllMovies(int page,int size,String sortBy,String sortDir);//get all movies
    MovieResponseDto getMovieById(Long id); //get movies by id
    void deleteMovie(long id);//delete movie
    MovieResponseDto updateMovie(Long id,MovieRequestDto movieRequestDto); //update the movie
    MovieResponseDto activateMovie(Long id);
      List<MovieResponseDto>getAllMovie(); //for all the database movies get
   //searching methods
    List<MovieResponseDto>searchMovie(String title);
    List<MovieResponseDto>getMoviesByGenre(String genre);
    List<MovieResponseDto>getMoviesByLanguage(String language);
    List<MovieResponseDto>getActiveMovies();
    List<MovieResponseDto>getUpcomingMovies();
    List<MovieResponseDto>getReleaseMovies();
    Long countActiveMovies();

}
